### DISCUSSION TITLE
Notification Drawer Enhancements

### DISCUSSION THREAD

**Discussion by w0kyj** [directus/directus#18707] [Feature Requests]
### Summary

Updates to the notifications drawer:
- Silence the number showing on top of the notifications icon
    - with an is_read flag after a user has read a notification(on click, toggle, ...)
    - or with a nullable date_read column
- The current notifications drawer was built to work for mentions in comments, but has never been extended beyond that. Now that we're allowing more generic notifications, we should add some sort of "detail view" for a given message (or maybe just expand it inline)
    - From https://github.com/directus/directus/pull/12522#issuecomment-1133015680

### Basic Example

Add any necessary additional fields to directus_notifications:
- Current notification item:
```json
{
"id": 67,
"timestamp": "2022-08-25T15:19:51.025Z",
"status": "inbox",
"recipient": "ba4b46bc-6147-47d0-9e66-cbe93328274c",
"sender": "ba4b46bc-6147-47d0-9e66-cbe93328274c",
"subject": "You were mentioned in articles",
"message": "\nHello Admin User,\n\nAdmin User has mentioned you in a comment:\n\n>  <em>Admin User</em> please review\n\n<a href=\"https://sales-demo-1.directus.app/admin/content/articles/3\">Click here to view.</a>\n",
"collection": "articles",
"item": "3"
}
```

Possible new notification example:
```json
{
"id": 67,
"timestamp": "2022-08-25T15:19:51.025Z",
"status": "inbox",
"recipient": "ba4b46bc-6147-47d0-9e66-cbe93328274c",
"sender": "ba4b46bc-6147-47d0-9e66-cbe93328274c",
"subject": "You were mentioned in articles",
"message": "\nHello Admin User,\n\nAdmin User has mentioned you in a comment:\n\n>  <em>Admin User</em> please review\n\n<a href=\"https://sales-demo-1.directus.app/admin/content/articles/3\">Click here to view.</a>\n",
"collection": "articles",
"item": "3",
"is_read": true,
"date_read": "2022-09-04T08:10:00.025Z"
}
```

### Motivation

The current notifications drawer for Studio App users doesn't show the "message"(comment or notification operation message) or provide a way to mark a notification read.

Users can only click the link or archive the message.

Adding a way to view the message body and mark them read provides a better user experience and increases adoption of using the Studio App.

### Detailed Design

Update the Notification Drawer API and Studio App UX to:
- Allow users to mark notification as read or unread:
    - Clicking a link to the collection/item should auto mark it read
    - A read message should not be included in the Notification Bell count
    - Support changing a read item to unread
    - Support bulk operations to mark multiple items read or unread
- Add a way to view the notification "message" details
    - inline dropdown or open item detail drawer
    - 

### Must Have:
- Allow users to mark notification as read or unread:
    - Clicking a link to the collection/item should auto mark it read
    - A read message should not be included in the Notification Bell count
    - Support changing a read item to unread
- Add a way to view the notification "message" details
    - inline dropdown or open item detail drawer
- Any new values must be populated with migrations or allow null values for backward compatibility of existing notifications.

### Should Have:
- Support bulk operations to mark multiple items read or unread

### Could Have:
- Forward notification
- Reply to sender

### Won't Have:
- 

### Drawbacks

Only real drawback is the cost and time to implement.

### Alternatives

Custom module
External Systems/Integrations

Current alternatives generally have users working around or outside of the Directus Studio app.

### Adoption Strategy

- No changes for Administrators
- Content editors will gain improved workflow and better collaboration with other users
- Any migrations would just populate current active notifications as unread
- No breaking changes expected

### Unresolved Questions

n/a

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
